sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2799,6 +2799,26 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
     }
   }
 
+  test("SPARK-24900: speed up sort when the dataset is small") {
```

Comment:
We may need unit test for `RangePartitioner` to test it correctly works with this change.

Reply:
UT for RangePartitioner has been added.